lib/node_modules/@stdlib/stats/incr/nanmmeanvar/test/test.js

```diff
@@ -0,0 +1,413 @@
+		actual = acc( data[ i ] );
+		t.deepEqual( actual, expected[ i ], 'returns expected value' );
+	}
+	t.end();
+});
+
+tape( 'the accumulator function computes a moving arithmetic mean and unbiased sample variance incrementally (output)', function test( t ) {
+	var expected;
+	var actual;
+	var data;
+	var acc;
+	var out;
+	var N;
+	var i;
+
+	data = [ 2.0, 3.0, 4.0, -1.0, 3.0, 1.0 ];
+	N = data.length;
+
+	out = [ 0.0, 0.0 ];
+	acc = incrnanmmeanvar( out, 3 );
+
+	expected = [
+		[ 2.0, 0.0 ],
+		[ 2.5, 0.5 ],
+		[ 3.0, 1.0 ],
+		[ 2.0, 7.0 ],
+		[ 2.0, 7.0 ],
+		[ 1.0, 4.0 ]
+	];
+	for ( i = 0; i < N; i++ ) {
+		actual = acc( data[ i ] );
+		t.strictEqual( actual, out, 'returns expected value' );
+		t.deepEqual( actual, expected[ i ], 'returns expected value' );
+	}
+	t.end();
+});
+
+tape( 'if not provided an input value, the accumulator function returns the current arithmetic mean and unbiased sample variance', function test( t ) {
+	var expected;
+	var actual;
+	var delta;
+	var data;
+	var tol;
+	var acc;
+	var i;
+
+	data = [ 2.0, 3.0, 10.0 ];
+	acc = incrnanmmeanvar( 3 );
+	for ( i = 0; i < data.length-1; i++ ) {
+		acc( data[ i ] );
+	}
+	t.deepEqual( acc(), [ 2.5, 0.5 ], 'returns expected value' );
+
+	acc( data[ data.length-1 ] );
+
+	expected = [ 5.0, 19.0 ];
+	actual = acc();
+
+	t.strictEqual( actual[ 0 ], expected[ 0 ], 'returns expected value' );
+
+	delta = abs( actual[ 1 ] - expected[ 1 ] );
+	tol = EPS * expected[ 1 ];
+	t.strictEqual( delta < tol, true, 'expected: '+expected[ 1 ]+'. actual: '+actual+'. tol: '+tol+'. delta: '+delta+'.' );
+
+	t.end();
+});
+
+tape( 'if data has yet to be provided, the accumulator function returns `null`', function test( t ) {
+	var acc = incrnanmmeanvar( 3 );
+	t.strictEqual( acc(), null, 'returns expected value' );
+	t.end();
+});
+
+tape( 'if only one datum has been provided, the accumulator function returns `0` for the sample variance', function test( t ) {
+	var acc = incrnanmmeanvar( 3 );
+	var v = acc( 2.0 );
+	t.strictEqual( v[ 0 ], 2.0, 'returns expected value' );
+	t.strictEqual( v[ 1 ], 0.0, 'returns expected value' );
+	t.end();
+});
+
+tape( 'if the window size is `1`, the accumulator functions always returns `0` for the sample variance', function test( t ) {
+	var acc;
+	var out;
+	var v;
+	var i;
+
+	acc = incrnanmmeanvar( 1 );
+	for ( i = 0; i < 100; i++ ) {
+		v = randu() * 100.0;
+		out = acc( v );
+		t.strictEqual( out[ 0 ], v, 'returns expected value' );
+		t.strictEqual( out[ 1 ], 0.0, 'returns expected value' );
+	}
+	t.end();
+});
+
+tape( 'if provided `NaN`, the accumulator skips the value and returns the previous mean and variance', function test( t ) {
+	var expected;
+	var data;
+	var acc;
+	var v;
+	var i;
+
+	acc = incrnanmmeanvar( 3 );
+
+	data = [
+		NaN,
+		3.14,
+		3.14,
+		NaN,
+		3.14,
+		3.14,
+		3.14,
+		NaN,
+		3.14,
+		3.14,
+		3.14,
+		NaN,
+		3.14,
+		3.14,
+		NaN,
+		NaN,
+		NaN,
+		NaN,
+		3.14
+	];
+	expected = [
+		null,
+		[ 3.14, 0.0 ],
```

Comment:
This and the following tests should be updated to be more meaningful in terms of actual computation.